File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        nums = []
        for number in numbers:
            nums.append(number)
        print(nums)

        for num in nums:
            for i in nums:
                if num + i == target:
                    return [nums.index(num)+1,nums.index(i)+1]

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number}
     */
    subarraySum(nums, k) {
        let res = 0;
        let curSum = 0;

        let prefixSum = new Map(); // sum -> cnt;
        prefixSum.set(0, 1); // sum of 0 found once;

        for (let num of nums) {
            curSum += num;
            let diff = curSum - k;
            res += prefixSum.get(diff) || 0;
            prefixSum.set(curSum, (prefixSum.get(curSum) || 0 ) + 1);
        }
        return res
    }
}
